x-pack/plugins/canvas/public/state/middleware/aeroelastic.js

```diff
@@ -66,10 +66,16 @@ const elementToShape = (element, i) => {
   };
 };
 
-const updateGlobalPositions = (setMultiplePositions, { shapes, gestureEnd }, elems) => {
+const updateGlobalPositions = (setMultiplePositions, { shapes, gestureEnd }, unsortedElements) => {
+  const ascending = (a, b) => (a.id < b.id ? -1 : 1);
+  const relevant = s => s.type !== 'annotation' && s.subtype !== 'adHocGroup';
+  const elements = unsortedElements.filter(relevant).sort(ascending);
```

Comment:
This could be turned into a lookup `{}` map though there'd be no discernible perf difference. Changes are kept streamlined in this PR due to planning to backport (minimal fix)